tests/runtime/test_caikit_health_probe.py

```diff
@@ -93,52 +98,71 @@ def temp_probe_config(*args, **kwargs):
-            # If booting the gRPC server, do so
-            with maybe_runtime_grpc_test_server(grpc_port):
-                # If only running gRPC, health probe should pass
-                assert caikit_health_probe.health_probe() == (
-                    server_mode == ServerMode.GRPC
-                )
-                # If booting the HTTP server, do so
-                with maybe_runtime_http_test_server(
-                    http_port, tls_config_override=config_overrides
-                ):
-                    # Probe should always pass with both possible servers up
-                    assert caikit_health_probe.health_probe()
+    with alog.ContextLog(
+        log.info,
+        "---LOG CONFIG: tls: %s, inline: %s, server: %s---",
+        tls_mode.name,
+        inline,
+        server_mode.name,
+    ):
+        # Get ports for both servers
+        http_port = tls_test_tools.open_port()
+        grpc_port = tls_test_tools.open_port()
+
+        # Set up SAN lists if not putting "localhost" in
+        server_sans, client_sans = None, None
+        if "NO_LOCALHOST" in tls_mode.name:
+            server_sans = ["foo.bar"]
+            client_sans = ["baz.bat"]
+
+        # Set up tls values if needed
+        tls = tls_mode == TlsMode.TLS
+        mtls = "MTLS" in tls_mode.name
+        with generate_tls_configs(
+            port=http_port,
+            tls=tls,
+            mtls=mtls,
+            inline=inline,
+            separate_client_ca=tls_mode == TlsMode.MTLS_SEPARATE_CLIENT_CA,
+            server_sans=server_sans,
+            client_sans=client_sans,
+        ) as config_overrides:
+            with temp_probe_config(
+                {
+                    "runtime": {
+                        "grpc": {
+                            "port": grpc_port,
+                            "enabled": server_mode
+                            in [ServerMode.GRPC, ServerMode.BOTH],
+                        },
+                        "http": {
+                            "enabled": server_mode
+                            in [ServerMode.HTTP, ServerMode.BOTH],
+                        },
+                    }
+                },
+                "merge",
+            ):
+                # Health probe fails with no servers booted
+                assert not caikit_health_probe.health_probe()
+                # If booting the gRPC server, do so
+                with maybe_runtime_grpc_test_server(grpc_port):
+                    # If only running gRPC, health probe should pass
+                    assert caikit_health_probe.health_probe() == (
+                        server_mode == ServerMode.GRPC
+                    )
+                    # If booting the HTTP server, do so
+                    with maybe_runtime_http_test_server(
+                        http_port, tls_config_override=config_overrides
+                    ):
+                        # Probe should always pass with both possible servers up
```

Comment:
Is there no test for what happens when there's no server up?

Reply:
https://github.com/caikit/caikit/pull/565/files#diff-7b05366351799f2ca03e0507a6bf16c4b2f9b1e47c3535f45cdcd5bffe0c7500R155